core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -106,7 +114,10 @@ public FastAppend appendManifest(ManifestFile manifest) {
       throw new RuntimeIOException(e, "Failed to write manifest");
     }
 
-    newManifests.addAll(appendManifests);
+    List<ManifestFile> appendManifestsWithMetadata = appendManifests.stream()
```

Comment:
I remember we prefer Guava transforms instead of the Stream API. I forgot the reason why.

Reply:
Either one can be reasonable. Sometimes streams are needlessly verbose, which is when I prefer Guava.